src/v/raft/tests/persisted_stm_test.cc

```diff
@@ -834,8 +834,18 @@ TEST_F_CORO(persisted_stm_test_fixture, test_snapshot_in_background_apply) {
     for (const auto& [_, stm] : node_stms) {
         ASSERT_EQ_CORO(stm->state, expected);
     }
-    // force background apply on one of the state machines
-    auto throwing_stm = node_stms.begin()->second;
+
+    // force background apply on one of the non-leader state machines
```

Comment:
If we are waiting for the commands to be applied on the leader in `apply_operations`, why then we need to wait for committed offset (line 848/858)? Presumably, everything should already be committed, no?